File: backend/dump_data.js
require('dotenv').config();

const fs = require('fs');
const { MongoClient } = require('mongodb');

async function dumpDataToDB() {

    const data = fs.readFileSync('../text_extraction/records.json');
    const jsonData = JSON.parse(data);

    // Connect to MongoDB
    const client = new MongoClient(process.env.MONGO_URI);
    try {
        await client.connect();

        // Get a reference to the database
        const database = client.db('trendall_records');

        // Get a reference to the collection
        const collection = database.collection('records');

        // Insert the data into the collection
        const result = await collection.insertMany(jsonData);
        console.log(`${result.insertedCount} documents inserted`);

    } finally {
        // Close the connection
        await client.close();
    }
}

// Call the function to dump the data into MongoDB
dumpDataToDB().catch(console.error);

File: text_extraction/dump_data.py
import json
from pymongo import MongoClient

uri = "mongodb+srv://trendall_admin:trendall_admin@trendallproject.4hr05ar.mongodb.net/?retryWrites=true&w=majority&appName=TrendallProject"

# Create a MongoDB client
client = MongoClient(uri)

# Connect to database
db = client["trendall_records"]
collection = db["records"]

# Load data from the JSON file
with open("records.json") as f:
    data = json.load(f)

# Insert data into the collection
collection.insert_many(data)
